rest_service/src/resources/messages.py

```diff
@@ -197,12 +216,12 @@ async def get_messages(
 )
 async def patch_message(
     message_update: dto_models.ReceivedMessageUpdate,
-    auth: refs.UserAuthProto = fastapi.Depends(refs.AuthGetterProto),
+    auth: dto_models.AuthUser = fastapi.Depends(refs.UserAuthProto),
     database: sql_api.DatabaseHandler = fastapi.Depends(refs.DatabaseProto),
     metadata: utilities.Metadata = fastapi.Depends(utilities.Metadata),
-    stored_message: dao_protos.Message = fastapi.Depends(retrieve_message),
+    stored_message: dao_protos.Message = fastapi.Depends(user_auth_message),
 ) -> dto_models.Message:
-    if stored_message.user_id != auth.user.id:
```

Comment:
Todo: this could actually be made a part or the sql interface to lower the amount of calls we have to make to it (aka itll only update it if both the id and user id match) if we`re sticking with this only the author can patch approach.
This approach could probably be used in other places as well